### PR TITLE
Show date time

### PR DESCRIPTION
**Pull request by Athozus**
This merge request allows you to see the date and the time while reading the message. It is based on `os.date()` native function which convert the epoch stored in the `.json` into a specific string, here `%Y-%m-%d %X`.

Related to #19 
screenshot_20230223_124449

### CODE REVIEW

**Comment by S-S-X** [comment]
Closes #19

**Review by fluxionary — COMMENTED** [review]
image

the timestamp runs a bit over the "close message" button, probably because i've got my font size increased slightly from the default.

IMO it'd be better if the timestamp was aligned w/ the rest of the headers.

File: gui.lua
Comment (S-S-X) [inline gui.lua:276]: There's been some issues caused by invalid or missing fields.
Not sure if this might be still issue or if it is fixed completely, issue might possible be there if "database" is very old?

To make it safe could make sure that  message.time is number (os.date will crash if not except for nil which returns current time):
```suggestion
	local date = type(message.time) == "number"
		and minetest.formspec_escape(os.date("%Y-%m-%d %X", message.time)) or ""
```

**Comment by Athozus** [comment]
Ok, thanks for replies.
Concerning the alignment of the date, I made it showing on the right to use white space, and I wanted to keep as possible the most place for the message itself.

**Comment by S-S-X** [comment]
> Concerning the alignment of the date, I made it showing on the right to use white space, and I wanted to keep as possible the most place for the message itself.

Looking at those screenshots I think you could compress fields a bit. Spacing even seems inconsistent currently.
I think spacing could be bit tighter, made consistent and timestamp added there.
All fields could also be bit closer to top edge to match left margin and make it bit more consistent with [x] button.

It anyway seems to eat near 30% of horizontal space with regular and about half with large font so I guess it would be best aligned to the left with other stuff.

**Comment by S-S-X** [comment]
I think another option would be to right align timestamp and clamp it to [x] button, also it does not really need that "Date: " caption.

**Comment by Athozus** [comment]
I understand what you're saying @S-S-X , but then I think it would be better to do as @fluxionary suggested, so all fields in vertical, but then add backgrounds colors to separate the different components (header & message) ?

**Comment by S-S-X** [comment]
> I understand what you're saying @S-S-X , but then I think it would be better to do as @fluxionary suggested, so all fields in vertical, but then add backgrounds colors to separate the different components (header & message) ?

Yes that's basically exactly what I also told in first response, just additionally explained how you can save some space by tightening and making spacing but layout outcome would be exactly what fluxionary suggested.

Second suggestion with aligning text to the right is option for layout you already used and that wont use layout suggested by fluxionray.
These are 2 different options I could think of first utilizing fluxionary's suggestion and second utilizing your original suggestion.

**Comment by S-S-X** [comment]
Well, I've tried right alignment and first test pretty much failed. This is how it looks like after changing font size (it apparently wont obey configuration, but yeah it is marked as experimental feature anyway...):
image

**Comment by Athozus** [comment]
Okay so I've just made an optimized header (in my opinion, maybe not yours) with a default MT green background, so I moved date to the bottom as suggested.

@S-S-X and @fluxionary , maybe keep that one ?
screenshot_20230226_000949

**Comment by S-S-X** [comment]
IMO looks good if spacing isn't too small for a bit larger font. 👍 

Now all you could ever want is just player avatar and mail domains to make it look almost like a real modern mail client 🤣 

image

**Comment by Athozus** [comment]
Lol, that's an idea :joy:. Anyway, showing avatar could be good 😁 , but I think it's not concerning a lot "show date". Maybe you should anyway share your commit, or I can rework the margins/paddings, or waiting for flux approval to merge ?

**Review by S-S-X — APPROVED** [review]
LGTM but maybe wait a moment if @fluxionary wants to test if a bit larger font is fine too or if spacing should be adjusted a bit.

**Review by fluxionary — APPROVED** [review]
LGTM

**Comment by BuckarooBanzay** [comment]
nice work :+1:

**Pull request merged**
